File: semester_3/WebLabs/Lab4/backend/src/main/java/com/joulin/Lab4/security/sevices/AuthUsersDetailsService.java
package com.joulin.Lab4.security.sevices;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import com.joulin.Lab4.entities.UserEntity;
import com.joulin.Lab4.repositories.UserRepository;

import javax.transaction.Transactional;
import java.util.Optional;

@Service
public class AuthUsersDetailsService implements UserDetailsService {

    private final UserRepository userRepository;

    @Autowired
    public AuthUsersDetailsService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    @Transactional
    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        Optional<UserEntity> user = userRepository.findByUsername(username);

        if (!user.isPresent())
            throw new UsernameNotFoundException("User Not Found with username: " + username);

        return AuthUsersDetails.build(user.get());
    }

}

File: semester_6/BLPS/Lab03/src/main/kotlin/su/arlet/business1/security/services/AuthUsersDetailsService.kt
package su.arlet.business1.security.services

import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

import su.arlet.business1.exceptions.UserNotFoundException
import su.arlet.business1.repos.UserRepo

@Service
class AuthUsersDetailsService(
    private val userRepo: UserRepo,
) : UserDetailsService {
    @Throws(UserNotFoundException::class)
    override fun loadUserByUsername(username: String): AuthUsersDetails {
        val user = userRepo.findByUsername(username)

        if (user.isEmpty) throw UsernameNotFoundException("User Not Found with username: $username")

        return AuthUsersDetails.build(user.get())
    }
}